src/Network/GRPC/Spec/LengthPrefixed.hs

```diff
@@ -121,8 +121,12 @@ buildMsg build compr (envelope, x) = mconcat [
     shouldCompress = and [
           not $ compressionIsIdentity compr
         , outboundEnableCompression envelope
-        , BS.Lazy.length compressed < BS.Lazy.length uncompressed
+        , uncompressedSizeThreshold compr uncompressedLength
+        , compressedLength < uncompressedLength
```

Comment:
Let's make this (1) threshold check (2) is enabled? (3) length (and skip the "is identity").